index.js

```diff
@@ -30,6 +30,56 @@ const getHeadSha = async (repo) => {
   return data.commit.sha;
 }
 
+async function closePRs() {
+   // Close old auto PRs
+   const {data: prs} = await octokit.pulls.list({
+     owner: 'cds-snc',
+     repo: 'notification-tf',
+     state: 'open'
+   });
+
+   prs.forEach( async pr => {
+     if(pr.title.startsWith("[AUTO-PR]")) {
+       await octokit.pulls.update({
+         owner: 'cds-snc',
+         repo: 'notification-tf',
+         pull_number: pr.number,
+         state: "closed"
+       });
+       await octokit.git.deleteRef({
+         owner: 'cds-snc',
+         repo: 'notification-tf',
+         ref: `heads/${pr.head.ref}`
+       });
+     }
+   })
+}
+
+async function hasChangesInStaging() {
+    pullsSearchDefinition = {
+        owner: 'cds-snc',
+        sort: 'updated',
+        direction: 'desc',
+        state: 'closed',
+        page: 1
+    }
+    const prodPulls = await octokit.pulls.list({
+        ...pullsSearchDefinition,
+        repo: 'notification-tf',
+    })
+    const stagingPulls = await octokit.pulls.list({
+        ...pullsSearchDefinition,
+        repo: 'notification-staging-tf',
+    })
+
+    const mergedProdPulls = prodPulls['data'].filter(pr => pr.merged_at !== null);
+    const mergedStagingPulls = stagingPulls['data'].filter(pr => pr.merged_at !== null);
+
+    const prodLastProdPR = Date.parse(mergedProdPulls[0].merged_at);
+    const stagingLastPR = Date.parse(mergedStagingPulls[0].merged_at);
+    return stagingLastPR > prodLastProdPR;
```

Comment:
You're only checking merge pull-requests for and not commits made to `master`. Are we sure that we can't commit directly to master on these repos? If the `master` branch is protected on both repositories that's fine 💯